### DISCUSSION TITLE
Get interaction forces

### DISCUSSION THREAD

**Discussion by VeronicaG24** [InfinyTech3D/SofaUnity#79] [Help - Using SofaUnity]
Hello,
I'm still learning to use Unity/Sofa/plugins so please be patient with me. 😄
I have diligently followed all the examples, tutorials, and demos on your website (https://infinytech3d.com/sapapi-unity3d/). However, I'm still looking for an example to obtain the contact forces (vectors or matrices, it doesn't matter).

My goal is to make an object deformable and elastic. When it collides with another object, it should change its shape based on the contact forces detected.

If it is possible to use this plugin for my goal, could someone help me find an example/demo where I can see how to obtain the contact forces from a collision between two objects? Is there any C# script available where the contact forces are detected, printed, or used in the simulation?

Thank you so much in advance!

**Comment by RickYang1** [comment]
Hi
You can find the rowForces of type float[] and the forces of type vector3[] in the SofaMesh class. However, when I print out the values of forces, I notice that they have the same values along the x, y, and z axes. I haven’t found a solution yet.

**Reply by epernod** [reply]
Hi @RickYang1 ,
the forces you are accessing are the internal forces not the contact force. However, it is strange that the x,y,z values are the same for each point. Could you post your error with your script. Maybe in another discussion.

Best regards,

**Comment by epernod** [comment]
Hi @VeronicaG24 ,

Yes what you want to achieve is possible with this asset.
How did you modeled the deformable object? Are you editing the SOFA scene?

Regarding the Collision force values access and display, it is work in progress and will be available in the next release which is plan for next month.

Best regards

**Reply by RickYang1** [reply]
> Hi @VeronicaG24 , 你好 ，
> 
> Yes what you want to achieve is possible with this asset.是的，您想要实现的目标可以通过此资产实现。 How did you modeled the deformable object? Are you editing the SOFA scene? 你是如何对可变形物体进行建模的？你在剪辑沙发场景吗？
> 
> Regarding the Collision force values access and display, it is work in progress and will be available in the next release which is plan for next month.关于碰撞力值的访问和显示，该工作正在进行中，并将在计划于下个月发布的下一个版本中提供。
> 
> Best regards 此致

Hi， @epernod ,
Sorry to bother you, may I ask when the next release will be released this month?

**Reply by epernod** [reply]
Hi,

By the end of this week. We need to do more tests and update the doc.

Best regards

**Comment by epernod** [comment]
Hi,

sorry for the late answer. Are you still working on that?

Best regards,
